pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -557,22 +565,10 @@ private void doAddConsumingSegment(String segmentName)
     PartitionDedupMetadataManager partitionDedupMetadataManager =
         _tableDedupMetadataManager != null ? _tableDedupMetadataManager.getOrCreatePartitionManager(partitionGroupId)
             : null;
-    RealtimeSegmentDataManager realtimeSegmentDataManager;
-    try {
-      realtimeSegmentDataManager =
-          createRealtimeSegmentDataManager(zkMetadata, tableConfig, indexLoadingConfig, schema, llcSegmentName,
-              consumerCoordinator, partitionUpsertMetadataManager, partitionDedupMetadataManager,
-              _isTableReadyToConsumeData);
-    } catch (SegmentAlreadyConsumedException e) {
```

Comment:
why remove this? without it, we will not unblock OFFLINE -> CONSUMING helix threads for segments that are already in completed state

Reply:
Lock is applied when starting the consumer thread, so this part won't throw this exception any more. This exception is also changed to be checked exception so that it won't be missed